examples/mapping_GOES16_TrueColor.py

```diff
@@ -269,34 +270,23 @@
 ######################################################################
 # Plot with `Cartopy`:  Geostationary Projection
 # ----------------------------------------------
-# 
+#
 # The image above is not georeferenced. You can see the land and oceans, but we
-# do have enough information to draw state and country boundaries. From the
-# GOES-16 file, we can extract all the metadata needed to understand the map
-# projection. We will use this data to plot the image on a `Basemap`. The image
-# is in a [geostationary
+# do have enough information to draw state and country boundaries. We will use
+# the `metpy.io` package to obtain the projection information. We will then use
+# Cartopy to plot the image. The image is in a [geostationary
 # projection](https://proj4.org/operations/projections/geos.html?highlight=geostationary).
 
-C['goes_imager_projection']
-
-######################################################################
-#
-
-# Satellite height
-sat_h = C['goes_imager_projection'].perspective_point_height
-
-# Satellite longitude
-sat_lon = C['goes_imager_projection'].longitude_of_projection_origin
-
-# Satellite sweep
-sat_sweep = C['goes_imager_projection'].sweep_angle_axis
-
-semi_major = C['goes_imager_projection'].semi_major_axis
-semi_minor = C['goes_imager_projection'].semi_minor_axis
+# We'll use the `CMI_C02` variable as a 'hook' to get the CF metadata.
+dat = C.metpy.parse_cf('CMI_C02')
+geos = dat.metpy.cartopy_crs
 
 # The projection x and y coordinates equals the scanning angle (in radians)
 # multiplied by the satellite height See details here:
 # https://proj4.org/operations/projections/geos.html?highlight=geostationary
+
+sat_h = C['goes_imager_projection'].perspective_point_height
+
 x = C['x'][:] * sat_h
```

Comment:
If you use `dat.x` and `dat.y`, MetPy will already have taken care of multiplying them by the satellite height for you. 😉 We try to make our users' lives as easy as we can.

Reply:
Out of curiosity, how did you do this? Just tons of if then else logic for various supported projections? Or maybe, the problem is not as unwieldy as I imagine it is.

Reply:
In every other CF-compliant projection, the coordinates have a dimensionality of "length", so we can use the unit handling support to change to meters and we're done. We then have a fallback path for the case where the coordinates are in radians and we do the multiplication by the `perspective_point_height`.

Reply:
OK, that worked.